design/authenticators/authn_jwt/token_schema.md

```diff
@@ -0,0 +1,178 @@
+
+### User Interface
+
+#### Authenticator Policy
+
+Two new variables will be added to the authenticator policy:
+
+* mandatory_claims
+* claims_mapping
+
+```yaml
+    - !policy
+    id: conjur/authn-jwt/VendorY
+    body:
+    - !webservice
+    
+    - !variable
+      id: mandatory_claims
+      
+    - !variable
+      id: claims_mapping
+```
+
+#### Variable Values Example
+
+| Name             | Value                   | Description                                      |
+| ---------------- | ----------------------- | ------------------------------------------------ |
+| mandatory_claims | branch, job, aud        | List of claims as they appear in host annotation |
+| claims_mapping   | {ref: branch, sub: job} | Mapping between host annotation to JWT claim     |
+
+* For any claim mapped, we will add the host annotation to the `mandatory_claims` variable and not the JWT claim name.
+* The values above are examples and they can be any claims.
+
+
+## Design
+[//]: # "Add any diagrams, charts and explanations about the design aspect of the solution. Elaborate also about the expected user experience for the feature"
+
+The following class diagram, represent the flow of checking the mandatory claims and the claim mapping.
+
+### Class Diagrams
+
+
+![image-20210711171238701](token-schema-class-diagram.png)
+
+### Flow Explanation
+
+When validate_restrictions is called in JWTVendorConfiguration class the following will happen as a result:
+
+1. The `valitate_restrictions` function will call `CreateConstraintsFromPolicy` that create constraints object checking the hosts annotations.
+   1. It will create instance of `NonEmptyConstraint` to check we have at least one host annotation.
+   2. It will call `LoadMAndatoryClaims` command class to load the list of mandatory claims for the `mandatory_claims` variable
+   3. It will create `RequiredContraint` object from it
+2. The `validate_restrictions` functon will call `CreateValidatorFromPolicy` to create validator with the claims mapping loaded
+   1. It will call the `LoadClaimsMapping` command class to get dictionary between claim to annotation.
+   2. It will create *ValidateRestrictionOneToOne* object and will return it
+3. `ValidateResourceRestrictions` will be called with the contraints and the validator from the previous steps. This run will check the claims in the token and annotations. If anything wrong like a missing claim in the host annotation or claim with invalid value and error with be thrown and the authentication request will be denied.
+
+## Backwards Compatibility
+[//]: # "How will the design of this solution impact backwards compatibility? Address how you are going to handle backwards compatibility, if necessary"
+
+Token schema is not mandatory therfore the JWT authentication will remain backwards compatible.
```

Comment:
not sure we can use 'token schema' prase anymore, this is now 'mandatory fields and mapped claims' feature